src/es_sfgtools/processing/assets/tiledb_temp.py

```diff
@@ -0,0 +1,188 @@
+    tiledb.Attr(name="pitch", dtype=np.float32),
+    tiledb.Attr(name="roll", dtype=np.float32),
+    attribute_dict["east"],
+    attribute_dict["north"],
+    attribute_dict["up"],
+    attribute_dict["east_std"],
+    attribute_dict["north_std"],
+    attribute_dict["up_std"],
+
+]
+PositionArraySchema = tiledb.ArraySchema(
+    sparse=True,
+    domain=tiledb.Domain(TimeDomain),
+    attrs=PositionAttributes,
+    cell_order='col-major', 
+    tile_order='row-major',
+    allows_duplicates=False,
+    coords_filters=filters,
+)
+
+ShotDataAttributes = [
+    tiledb.Attr(name="head0", dtype=np.float32),
+    tiledb.Attr(name="pitch0", dtype=np.float32),
+    tiledb.Attr(name="roll0", dtype=np.float32),
+    tiledb.Attr(name="head1", dtype=np.float32),
+    tiledb.Attr(name="pitch1", dtype=np.float32),
+    tiledb.Attr(name="roll1", dtype=np.float32),
+    tiledb.Attr(name="east0", dtype=np.float32),
+    tiledb.Attr(name="north0", dtype=np.float32),
+    tiledb.Attr(name="up0", dtype=np.float32),
+    tiledb.Attr(name="east1", dtype=np.float32),
+    tiledb.Attr(name="north1", dtype=np.float32),
+    tiledb.Attr(name="up1", dtype=np.float32),
+    attribute_dict["east_std"],
+    attribute_dict["north_std"],
+    attribute_dict["up_std"],
+]
+
+ShotDataArraySchema = tiledb.ArraySchema(
+    sparse=True,
+    domain=tiledb.Domain(TimeDomain),
+    attrs=ShotDataAttributes,
+    cell_order='col-major', 
+    tile_order='row-major',
+    allows_duplicates=False,
+    coords_filters=filters,
+)
+
+accousticTimeDim = tiledb.Dim(name="triggerTime", domain=(0,), tile=100, dtype="datetime64[ms]")
+acousticIDDim = tiledb.Dim(name="transponderID", domain=(0,), tile=100, dtype="U")
+
+AcousticDataAttributes = [
+    tiledb.Attr(name="transponderID",dtype="U"),
+    tiledb.Attr(name="pingTime",dtype=np.float32),
+    tiledb.Attr(name="returnTime",dtype=np.float32),
+    tiledb.Attr(name="tt",dtype=np.float32),
+    tiledb.Attr(name="dbv",dtype=np.uint8),
+    tiledb.Attr(name="xc",dtype=np.uint8),
+    tiledb.Attr(name="snr",dtype=np.float32),
+    tiledb.Attr(name="tat",dtype=np.float32),
+]
+AcousticArraySchema = tiledb.ArraySchema(
+    sparse=True,
+    domain=tiledb.Domain(accousticTimeDim,acousticIDDim),
+    attrs=AcousticDataAttributes,
+    cell_order='col-major', 
+    tile_order='row-major',
+    allows_duplicates=True,
+    coords_filters=filters,
+)
+
+
+class TDBAcousticArray:
+    def __init__(self,uri:Path|str):
+        if isinstance(uri,str):
+            uri = Path(uri)
+        self.uri = uri
+        if not uri.exists():
+            tiledb.Array.create(str(uri),AcousticArraySchema)
+  
+    @check_types
+    def write_df(self,df:DataFrame[AcousticDataFrame]):
+       tiledb.from_pandas(str(self.uri),df,mode='append')
+
+    @check_types
+    def read_df(self,start:datetime,end:datetime)->DataFrame[AcousticDataFrame]:
+        return tiledb.read_pandas(str(self.uri),start=start,end=end)
```

Comment:
I dont think read_pandas is a method?
I think you need to use a context manager to open the array, and then query and df ie
with tiledb.open(self.array.uri, "r", ctx=self.array.ctx) as A:
    df = A.query(index_col='time').df[start:end]